src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java

```diff
@@ -334,6 +310,162 @@ public CaseDTO getCaseByCaseReference(final long caseRef, CaseQueryRequestDTO re
     return caseServiceResponse;
   }
 
+  private void validateCompatibleEstabAndCaseType(CaseType caseType, EstabType estabType)
+      throws CTPException {
+    Optional<AddressType> addrType = estabType.getAddressType();
+    if (addrType.isPresent() && (caseType != CaseType.valueOf(addrType.get().name()))) {
+      log.with("caseType", caseType)
+          .with("estabType", estabType)
+          .info("Mismatching caseType and estabType");
+      String msg =
+          "Derived address type of '"
+              + addrType.get()
+              + "', from establishment type '"
+              + estabType
+              + "', "
+              + "is not compatible with caseType of '"
+              + caseType
+              + "'";
+      throw new CTPException(Fault.BAD_REQUEST, msg);
+    }
+  }
+
+  private boolean isAddressTypeChange(CaseType requestedCaseType, String existingEstabType) {
```

Comment:
Once the initial test has been done to assert that the new estabType.addressType is compatible with the new caseType, its use can be limited to just setting the estabType in the outgoing event. From that point on we should just consider the old and new caseTypes

Reply:
done